modeshape-persistence-tests/pom.xml

```diff
@@ -0,0 +1,332 @@
+                                </wait>
+                            </run>
+                            <build>
+                                <from>postgres</from>
+                            </build>
+                        </image>
+                    </images>
+                </configuration>
+                <!--
+                Connect this plugin to the maven lifecycle around the integration-test phase:
+                start the container in pre-integration-test and stop it in post-integration-test.
+                -->
+                <executions>
+                    <execution>
+                        <id>start</id>
+                        <phase>pre-integration-test</phase>
+                        <goals>
+                            <goal>build</goal>
+                            <goal>start</goal>
+                        </goals>
+                    </execution>
+                    <execution>
+                        <id>stop</id>
+                        <phase>post-integration-test</phase>
+                        <goals>
+                            <goal>stop</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            
+            
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-failsafe-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>integration-test-mysql</id>
+                        <goals>
+                            <goal>integration-test</goal>
+                        </goals>
+                        <configuration>
+                            <systemPropertyVariables>
+                            <!-- Make these available to the tests via system properties -->
+                                <database.driver>com.mysql.jdbc.Driver</database.driver>
+                                <database.url>jdbc:mysql://localhost:3306/${database.name}</database.url>
+                                <database.user>${database.user}</database.user>
+                                <database.password>${database.password}</database.password>
+                            </systemPropertyVariables>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>integration-test-postgres</id>
+                        <goals>
+                            <goal>integration-test</goal>
+                        </goals>
+                        <configuration>
+                            <systemPropertyVariables>
+                            <!-- Make these available to the tests via system properties -->
+                                <database.driver>org.postgresql.Driver</database.driver>
+                                <database.url>jdbc:postgresql://localhost/${database.name}</database.url>
```

Comment:
Using `localhost` will only work on Linux, and will not currently work on OS X or Windows because on those platforms Docker is run in a virtual machine. Doesn't the Fabric8 Maven plugin set the `${docker.host}` on all platforms, even when Docker on Linux doesn't set `DOCKER_HOST` env variable? If so, then this should probably be:

    <database.url>jdbc:postgresql://${docker.host}/${database.name}</database.url>